js/ui/grid_core/ui.grid_core.validating.js

```diff
@@ -177,12 +179,16 @@ const ValidatingController = modules.Controller.inherit((function() {
             const editMode = this._editingController.getEditMode();
 
             if(FORM_BASED_MODES.indexOf(editMode) === -1) {
+                if(editData.type === EDIT_DATA_INSERT_TYPE && (!isDefined(editData.data) || isEmptyObject(editData.data))) {
```

Comment:
`(!isDefined(editData.data) || isEmptyObject(editData.data))`
Why is such strange condition needed?

Reply:
isDefined is redundant here. So, the idea is to prevent the hidden validation from executing for a new unmodified row. I'll move the following condition
```js
isEmptyObject(editData.data)
```
to a separate method and call it isRowModified. So, it'll be:
```js
isRowModified(editData) {
    return !isEmptyObject(editData.data);
},
...
if(editData.type === EDIT_DATA_INSERT_TYPE && !this.isRowModified(editData)) {
    editData.isValid = true;
    return;
}
```